nginx-src/src/download.rs

```diff
@@ -0,0 +1,257 @@
+                "8657ABB260F056B1E5190839D9C4D26D0E604491",
+                "B7C1C14360F353A36862E4D5231C84CDDCC69C45",
+                "95A9908DDFA16830BE9FB9003D30A3A9FF1360DC",
+                "7953AC1FBC3DC8B3B292393ED5E9E43F7DF9EE8C",
+                "E5E52560DD91C556DDBDA5D02064C53641C25E5D",
+                "C1F33DD8CE1D4CC613AF14DA9195C48241FBF7DD",
+                "BA5473A2B0587B07FB27CF2D216094DFD0CB81EF",
+            ],
+        },
+    ),
+    (
+        "pcre",
+        SourceSpec {
+            url: |version| {
+                // We can distinguish pcre1/pcre2 by checking whether the second character is '.',
+                // because the final version of pcre1 is 8.45 and the first one of pcre2 is 10.00.
+                if version.chars().nth(1).is_some_and(|c| c == '.') {
+                    format!("{PCRE1_URL_PREFIX}/{version}/pcre-{version}.tar.gz")
+                } else {
+                    format!("{PCRE2_URL_PREFIX}/pcre2-{version}/pcre2-{version}.tar.gz")
+                }
+            },
+            variable: "PCRE2_VERSION",
+            signature: "sig",
+            keyserver: UBUNTU_KEYSEVER,
+            key_ids: &[
+                // Key 1: Phillip Hazel's public key. For PCRE2 10.44 and earlier
+                "45F68D54BBE23FB3039B46E59766E084FB0F43D8",
+                // Key 2: Nicholas Wilson's public key. For PCRE2 10.45
+                "A95536204A3BB489715231282A98E77EB6F24CA8",
+            ],
+        },
+    ),
+    (
+        "zlib",
+        SourceSpec {
+            url: |version| format!("{ZLIB_URL_PREFIX}/v{version}/zlib-{version}.tar.gz"),
+            variable: "ZLIB_VERSION",
+            signature: "asc",
+            keyserver: UBUNTU_KEYSEVER,
+            key_ids: &[
+                // Key 1: Mark Adler's public key. For zlib 1.3.1 and earlier
+                "5ED46A6721D365587791E2AA783FCD8E58BCAFBA",
+            ],
+        },
+    ),
+];
+
+static VERIFIER: LazyLock<Option<SignatureVerifier>> = LazyLock::new(|| {
+    SignatureVerifier::new()
+        .inspect_err(|err| eprintln!("GnuPG verifier: {err:?}"))
+        .ok()
+});
+
+fn make_cache_dir() -> io::Result<PathBuf> {
+    let base_dir = env::var("CARGO_MANIFEST_DIR")
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| env::current_dir().expect("Failed to get current directory"));
+    // Choose `.cache` relative to the manifest directory (nginx-src) as the default cache directory
+    // Environment variable `CACHE_DIR` overrides this
+    // Recommendation: set env "CACHE_DIR = { value = ".cache", relative = true }" in
+    // `.cargo/config.toml` in your project
+    let cache_dir = env::var("CACHE_DIR")
+        .map(PathBuf::from)
+        .unwrap_or(base_dir.join(".cache"));
+    if !cache_dir.exists() {
+        fs::create_dir_all(&cache_dir)?;
+    }
+    Ok(cache_dir)
+}
+
+/// Downloads a tarball from the specified URL into the `.cache` directory.
```

Comment:
Downloads, Get, Extract, Run, Returns   - grammar is a bit inconsistent in comments. 

I am fine both ways.  But it is better to be consistent.